public/app/features/dashboard-scene/serialization/transformSceneToSaveModelSchemaV2.ts

```diff
@@ -141,6 +145,76 @@ export function transformSceneToSaveModelSchemaV2(scene: DashboardScene, isSnaps
   }
 }
 
+function getLayout(
+  layoutManager: DashboardLayoutManager,
+  isSnapshot?: boolean
+): GridLayoutKind | RowsLayoutKind | ResponsiveGridLayoutKind {
+  if (layoutManager instanceof DefaultGridLayoutManager) {
+    return {
+      kind: 'GridLayout',
+      spec: {
+        items: getGridLayoutItems(layoutManager, isSnapshot),
+      },
+    };
+  } else if (layoutManager instanceof RowsLayoutManager) {
+    return {
+      kind: 'RowsLayout',
+      spec: {
+        rows: layoutManager.state.rows.map((row) => {
+          if (row.state.layout instanceof RowsLayoutManager) {
+            throw new Error('Nesting row layouts is not supported');
+          }
+          let layout: RowGridLayoutKind | ResponsiveGridLayoutKind | undefined = undefined;
+          if (row.state.layout instanceof DefaultGridLayoutManager) {
+            layout = getRowGridLayout(row.state.layout, isSnapshot);
+          } else if (row.state.layout instanceof ResponsiveGridLayoutManager) {
+            layout = {
+              kind: 'ResponsiveGridLayout',
+              spec: {
+                items: getResponsiveGridLayoutItems(row.state.layout),
+                col:
+                  row.state.layout.state.layout.state.templateColumns?.toString() ??
+                  'repeat(auto-fit, minmax(400px, auto))',
+                row: row.state.layout.state.layout.state.autoRows?.toString() ?? 'minmax(300px, auto)',
+              },
+            };
+          }
+          if (!layout) {
+            throw new Error('Unsupported layout type');
+          }
+          return {
+            kind: 'RowsLayoutRow',
+            spec: {
+              title: row.state.title,
+              collapsed: row.state.isCollapsed ?? false,
+              layout: layout,
+            },
+          };
+        }),
+      },
+    };
+  } else if (layoutManager instanceof ResponsiveGridLayoutManager) {
+    return {
+      kind: 'ResponsiveGridLayout',
+      spec: {
+        items: getResponsiveGridLayoutItems(layoutManager),
+        col: layoutManager.state.layout.state.templateColumns?.toString() ?? 'repeat(auto-fit, minmax(400px, auto))',
```

Comment:
I think we can extract the `repeat(auto-fit, minmax(400px, auto))` part to a constant and use it inside the `ResponsiveGridLayoutManager` as well to avoid mismatches if we ever change the default.

Reply:
Agree. Will work on a defaults here. Also the ResponsiveGridLayout schema probably needs some work. I only implemented as much as absolutely needed to support what could be changed in the UI